tapdb/sqlc/migrations/000017_seedling_script_group_keys.up.sql

```diff
@@ -0,0 +1,12 @@
+-- According to SQLite docs, a column added via ALTER TABLE cannot be both
+-- a REFERENCE and NOT NULL, so we'll have to enforce non-nilness outside of the DB.
+ALTER TABLE asset_seedlings ADD COLUMN script_key_id BIGINT REFERENCES script_keys(script_key_id);
+
+-- For a group anchor, we derive the internal key for the future group key early,
+-- to allow use of custom group witnesses.
+ALTER TABLE asset_seedlings ADD COLUMN group_internal_key_id BIGINT REFERENCES internal_keys(key_id);
+
+-- For a group key, the internal key can also be tweaked to commit to a
+-- tapscript tree. Once we finalize the batch, this tweak will also be stored
+-- as part of the asset group itself.
+ALTER TABLE asset_seedlings ADD COLUMN group_tapscript_root BLOB;
```

Comment:
Will this also eventually ref the tapscript trees we store on disk? 

Also wouldn't it be enough to just ref an group key here since that already have the sibling hash as a column?

Reply:
> Will this also eventually ref the tapscript trees we store on disk?

The value will, but this specific field won't. It'll be copied over to the final entry in the `asset_groups` table. I don't think we need to require a stored tapscript tree with a matching root though. If the user is generating a key-path witness, we only need the root and not the full tree. We only need the tree to validate a script-path witness.

"Make a group key with this tapscript tree root. If I want to produce a script-path witness, I'll later provide the tapscript tree for this tapscript tree root."

We haven't exposed the tapscript tree store yet, so the tapscript tree could be a field in the params for `sealBatch`.

> Also wouldn't it be enough to just ref an group key here since that already have the sibling hash as a column?

No, as this happens before group keys have been fully formed and inserted into `asset_groups` at all. Before being able to seal the batch, we need both an internal key for the group and this optional tweak; the asset ID can be computed after batch funding, which is the missing piece for deriving the final tweaked key.

We can only insert the asset group once we have the group witness, so at the end of `sealBatch`.